src/components/axes/timeAxis.ts

```diff
@@ -427,9 +397,10 @@ export module Axis {
         this._cleanTier(i);
       }
 
-      var tierTicks = tierConfigs.map((config: TimeAxisTierConfiguration, i: number) =>
-        this._renderTierLabels(this._tierLabelContainers[i], config, i)
-      );
+      var tierTicks = tierConfigs.map((config: TimeAxisTierConfiguration, i: number) => {
+        this._renderTierLabels(this._tierLabelContainers[i], config, i);
+        return this._getTickValuesForConfiguration(config);
+      });
```

Comment:
How much cleaner would it be to split this into a foreach and a map?  Not that big of a deal